### PR TITLE
Cell indexes in Row not being respected when rendering Row

### PR DESCRIPTION
**Pull request by Efiware**
See Issue #682

### CODE REVIEW

**Comment by boxcla** [comment]
Hi @Efiware, thanks for the pull request. Before we can merge it, we need you to sign our Contributor License Agreement. You can do so electronically here: http://opensource.box.com/cla

Once you have signed, just add a comment to this pull request saying, "CLA signed". Thanks!

**Comment by Efiware** [comment]
CLA signed

**Comment by boxcla** [comment]
Verified that @Efiware has just signed the CLA. Thanks, and we look forward to your contribution.

File: src/Spout/Writer/XLSX/Manager/WorksheetManager.php [hunk 1]
```diff
@@ -153,16 +153,17 @@ public function addRow(Worksheet $worksheet, Row $row)
...
-        $numCells = $row->getNumCells();
+        $numCells = max(array_keys($row->getCells()))+1;
 
         $rowXML = '<row r="' . $rowIndex . '" spans="1:' . $numCells . '">';
 
-        foreach ($row->getCells() as $cell) {
-            $rowXML .= $this->applyStyleAndGetCellXML($cell, $rowStyle, $rowIndex, $cellIndex);
-            $cellIndex++;
+        $cells = $row->getCells();
+        for ($cell_index=0; $cell_index<$numCells; $cell_index++) {
```
Comment (adrilo) [inline src/Spout/Writer/XLSX/Manager/WorksheetManager.php]: Please follow project code style.

File: src/Spout/Writer/XLSX/Manager/WorksheetManager.php [hunk 2]
```diff
@@ -153,16 +153,17 @@ public function addRow(Worksheet $worksheet, Row $row)
      */
     private function addNonEmptyRow(Worksheet $worksheet, Row $row)
     {
-        $cellIndex = 0;
         $rowStyle = $row->getStyle();
         $rowIndex = $worksheet->getLastWrittenRowIndex() + 1;
-        $numCells = $row->getNumCells();
+        $numCells = max(array_keys($row->getCells()))+1;
```
Comment (adrilo) [inline src/Spout/Writer/XLSX/Manager/WorksheetManager.php]: I'd rather change `getNumCells` to avoid similar issues elsewhere

**Comment by adrilo** [comment]
Please add a test.

**Pull request closed without merging**